search/query/disjunction.go

```diff
@@ -7,65 +7,55 @@
 //  either express or implied. See the License for the specific language governing permissions
 //  and limitations under the License.
 
-package bleve
+package query
 
 import (
 	"encoding/json"
+	"fmt"
 
 	"github.com/blevesearch/bleve/index"
+	"github.com/blevesearch/bleve/mapping"
 	"github.com/blevesearch/bleve/search"
 	"github.com/blevesearch/bleve/search/searchers"
 )
 
-type disjunctionQuery struct {
+type DisjunctionQuery struct {
 	Disjuncts []Query `json:"disjuncts"`
-	BoostVal  float64 `json:"boost,omitempty"`
-	MinVal    float64 `json:"min"`
+	Boost     *Boost  `json:"boost,omitempty"`
+	Min       float64 `json:"min"`
 }
 
 // NewDisjunctionQuery creates a new compound Query.
 // Result documents satisfy at least one Query.
-func NewDisjunctionQuery(disjuncts []Query) *disjunctionQuery {
-	return &disjunctionQuery{
+func NewDisjunctionQuery(disjuncts []Query) *DisjunctionQuery {
+	return &DisjunctionQuery{
 		Disjuncts: disjuncts,
-		BoostVal:  1.0,
 	}
 }
 
 // NewDisjunctionQueryMin creates a new compound Query.
 // Result documents satisfy at least min Queries.
-func NewDisjunctionQueryMin(disjuncts []Query, min float64) *disjunctionQuery {
-	return &disjunctionQuery{
+func NewDisjunctionQueryMin(disjuncts []Query, min float64) *DisjunctionQuery {
+	return &DisjunctionQuery{
 		Disjuncts: disjuncts,
-		BoostVal:  1.0,
-		MinVal:    min,
+		Min:       min,
 	}
 }
 
-func (q *disjunctionQuery) Boost() float64 {
-	return q.BoostVal
+func (q *DisjunctionQuery) SetBoost(b float64) {
+	boost := Boost(b)
+	q.Boost = &boost
 }
 
-func (q *disjunctionQuery) SetBoost(b float64) Query {
-	q.BoostVal = b
-	return q
-}
-
-func (q *disjunctionQuery) AddQuery(aq Query) Query {
+func (q *DisjunctionQuery) AddQuery(aq Query) {
```

Comment:
One thought is if this took the multi-value '...' idiom as an input type then folks could then use...

   disjQuery.AddQuery(q1, q2, q3)

or

   disjQuery.AddQuery(...queryParts)